rohmu/object_storage/s3.py

```diff
@@ -395,6 +402,7 @@ def multipart_upload_file_object(self, *, cache_control, fp, key, metadata, mime
         self.log.info(
             "Multipart upload of %r complete, size: %r, took: %.2fs", key, size, time.monotonic() - start_of_multipart_upload
         )
+        self.notify_write(key=key, size=self.get_file_size(plain_key))
```

Comment:
`get_file_size()` could fail here (also true for other remote storage drivers), should we handle that and allow a value for unknown file size (`-1` ?)

Reply:
Do you mean if the cloud provider has consistency issues? (Well, that is probably the case for Azure, their API is weirdly inconsistent)

Reply:
Also, but even just the fact it's ann HTTP HEAD request... chances of it failing should be quite low (especially if the transfer goes through :D) but still, it "could" (transient network hiccup)